doc/source/train/examples/pytorch/pytorch-profiling/README.md

```diff
@@ -0,0 +1,465 @@
+    
+    # Export memory timeline
+    prof.export_memory_timeline(
+        f"/mnt/cluster_storage/{run_name}/rank{world_rank}_advanced_memory_profile.html"
+    )
+    
+    
+    if world_rank == 0:
+        print(f"Advanced profiling complete! Check '/mnt/cluster_storage/{run_name}/' for detailed profiling results.")
+        print("Files generated:")
+        print(f"  - rank{world_rank}_advanced_memory_profile.html (Memory analysis)")
+        print(f"  - rank{world_rank}_chrome_trace.json (Chrome trace)")
+        print("  - TensorBoard logs in /mnt/cluster_storage/logs/advanced/")
+
+```
+
+Kick off the Ray Train job similarly to the previous step.
+
+
+```python
+# Run the advanced profiling example
+scaling_config = ray.train.ScalingConfig(num_workers=1, use_gpu=True)
+
+# Create a unique experiment name for advanced profiling
+advanced_experiment_name = f"advanced_profiling_{uuid.uuid4().hex[:8]}"
+
+# Configure run settings
+run_config = ray.train.RunConfig(
+    storage_path="/mnt/cluster_storage/",
+    name=advanced_experiment_name,
+)
+
+# Launch advanced profiling training job
+trainer = ray.train.torch.TorchTrainer(
+    train_func_advanced_profiling,
+    scaling_config=scaling_config,
+    run_config=run_config,
+)
+
+print(f"Starting advanced profiling training: {advanced_experiment_name}")
+result = trainer.fit()
+print("Advanced profiling training completed successfully!")
+print(f"Check '/mnt/cluster_storage/{advanced_experiment_name}/' for comprehensive profiling results.")
+
+```
+
+### Analyzing profiling results
+
+After running the profiling examples, you have access to several types of profiling data:
+
+1. **TensorBoard traces**: Located in `/mnt/cluster_storage/logs/` or the persistent storage that configured. - Use these traces to visualize GPU/CPU utilization, kernel execution times, and memory allocation patterns.
+
+2. **Memory timeline HTML files**: Worker-specific memory profiles showing memory usage over time, helping identify memory leaks and optimization opportunities.
+
+
+3. **Ray Train dashboard**: If using Anyscale workspace, access the Ray Train dashboard to monitor real-time metrics, worker status, and resource utilization.
+
+### Key profiling insights to look for:
+
+- **GPU utilization**: Ensure your workload is using GPUs efficiently (high utilization percentage)
+- **Memory usage patterns**: Look for memory spikes, leaks, or inefficient allocation patterns
+- **Communication overhead**: Monitor time spent on gradient synchronization
+- **Data loading bottlenecks**: Identify if data loading is limiting training throughput
+- **Kernel efficiency**: Analyze which operations are taking the most time and optimize accordingly
+
+
+### Tensorboard plugin with PyTorch Profiler and tracing.
+
+After generating the `trace.json` files, you can use Tensorboard, or drag the `trace.json` into Perfetto UI or `chrome://tracing` to visualize your profile.
```

Comment:
Earlier you mentioned that these were the files generated

print("Files generated:")
print(f"  - rank{world_rank}_advanced_memory_profile.html (Memory analysis)")
print(f"  - rank{world_rank}_chrome_trace.json (Chrome trace)")
print("  - TensorBoard logs in /mnt/cluster_storage/logs/advanced/")

Are all your screenshots from Tensorboard? Which of these files is Tensorboard using? It's probably worth clarifying that here since from reading this sentence it sounds like `trace.json` is responsible for all of this information.